snappi_ixnetwork/trafficitem.py

```diff
@@ -2157,4 +2157,277 @@ def rocev2_flow_results(self, request):
+
+            tr["trafficItem"][-1]["endpointSet"] = [
+                {
+                    "xpath": tr["trafficItem"][-1]["xpath"]
+                    + "/endpointSet[1]",
+                }
+            ]
+            if flow.tx_rx.choice == "port":
+                tr_type = "raw"
+                ep = getattr(flow.tx_rx, "port")
+                tx_objs = ["%s/protocols" % ports.get(ep.tx_name)]
+                rx_objs = ["%s/protocols" % ports.get(ep.rx_name)]
+                tr["trafficItem"][-1]["endpointSet"][0]["sources"] = [
+                    o for o in tx_objs
+                ]
+                tr["trafficItem"][-1]["endpointSet"][0]["destinations"] = [
+                    o for o in rx_objs
+                ]
+            else:
+                ep = getattr(flow.tx_rx, "device")
+                tr_type = devices[ep.tx_names[0]]["type"]
+                source = []
+                destinations = []
+                scalable_sources = []
+                scalable_destinations = []
+                self._gen_dev_endpoint(
+                    devices, ep.tx_names, source, scalable_sources
+                )
+                self._gen_dev_endpoint(
+                    devices, ep.rx_names, destinations, scalable_destinations
+                )
+                if len(source) > 0:
+                    tr["trafficItem"][-1]["endpointSet"][0]["sources"] = source
+                if len(destinations) > 0:
+                    tr["trafficItem"][-1]["endpointSet"][0][
+                        "destinations"
+                    ] = destinations
+                if len(scalable_sources) > 0:
+                    tr["trafficItem"][-1]["endpointSet"][0][
+                        "scalableSources"
+                    ] = scalable_sources
+                if len(scalable_destinations) > 0:
+                    tr["trafficItem"][-1]["endpointSet"][0][
+                        "scalableDestinations"
+                    ] = scalable_destinations
+
+            tr["trafficItem"][-1]["trafficType"] = tr_type
+            if tr_type == "raw":
+                tr["trafficItem"][-1]["configElement"] = self.config_raw_stack(
+                    tr_xpath, self._flows_packet[index]
+                )
+            self.traffic_index += 1
+            self.logger.debug(
+                "Flow %s converted to %s" % (flow_name, tr["trafficItem"][-1])
+            )
+        return tr
+
+    def _validate_append_flows_config(self, append_flows_config):
+        initial_flownames = []
+        for i_flow in self._api._config.flows._items:
+            initial_flownames.append(i_flow.name)
+        errors = []
+        for flow in append_flows_config:
+            if flow.name in initial_flownames:
+                errors.append(
+                    "flow {} is already present in the configuration".format(
+                        flow.name
+                    )
+                )
+            self._api._config.flows._items.append(flow)
```

Comment:
probably we need to put it into else block otherwise that flow actually added though we have error on it.